specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json

```diff
@@ -0,0 +1,247 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ComputeDiagnosticResourceProviderClient",
+    "description": "The Compute Diagnostic Resource Provider Client",
+    "version": "2021-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/DiagnosticOperations/{operationId}": {
+      "get": {
+        "tags": [
+          "DiagnosticOperations"
+        ],
+        "operationId": "DiagnosticOperations_Read",
+        "description": "Gets status of a Diagnostic operation issued for Disk Inspection.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "in": "path",
+            "name": "operationId",
+            "required": true,
+            "type": "string",
+            "description": "The identifier pointing to a request for Disk Inspection. This is used for polling result of the request."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/Diagnostics/runDiskInspection": {
```

Comment:
POST must not be used on a collection of resources. POST should be used on a single resource like below
/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/{diagnosticId}/runDiskInspection

Refer best practice 'POST on collection segments' in https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html

Reply:
Thanks, Ramyasree! We are aware of that for Azure 1P RPs, POST is usually an operational task on an existing resource and PUT is the initial create operation (i.e. PUT VM). And this is keeping in mind that PUT and POST have always been somewhat ambiguous even outside the context of Azure, i.e. "https://restfulapi.net/rest-put-vs-post/". The RFC definition seems somewhat different from the way Azure is interpreting this too: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6.  

For our reasoning, we had thought that the Azure interpretation of POST vs PUT would **only** apply to ARM tracked resource types and we didn't know it would apply for Proxy-Only resource types such as ours as well. (It _is_ a bit confusing since I think we can probably find a few other existing Proxy-Only resource types that accept POST without a singular resource Id reference, but we don't have to digress and go down that path :) ) For our resource type, since each of our diagnostic run is a discrete operational task on its own where no ARM tracked resource is created, we had opted for the POST verb with no tracking unique identifier.

If you feel strongly that the same POST/PUT interpretation should apply to Proxy-Only resource types as well, I think we would prefer to replace the verb with PUT from POST. Asking the caller or client to generate a unique identifier for a diagnostic run is a bigger change and it makes the call feel a lot more like an ARM tracked resource. 

Please let us know what you think.

Reply:
POST guidance applies to proxy resources as well. Existing APIs that didn’t follow this guidance must have been introduced long time back. Lately ARM API review board is enforcing the guidelines strictly.

PUT doesn’t make sense in your scenario as runDiskInspection is an action. POST is the right verb to use. We can explore modeling the API something like below. Please setup a short meeting with me tomorrow to close on this.

POST /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/run

Reply:
I have updated swagger as per discussion -
1) Updated API for POST call
2) Added GET calls for Diagnostic collection and DiskInspection type.

Reply:
@ansahdev - I have added some comments. Once you resolve them, I will sign off.